### PR TITLE
fix: improve UI Kit boards

### PR DESCRIPTION
**Pull request by tetianaman**
[Figma ](https://www.figma.com/design/RXdC3QEf72EUey3NHq9arm/Ui-Kit?node-id=335-1618&node-type=frame&t=3OMAy0C48CJJXVLm-0) 

I created Kit component with UI Kit common elements and styles.

Notes:
1. No menu in "Buttons" board, because menu is a part of `Header` component.
2. No cards row in "Components & Elements" board, because we have no such a component, but we have `FeaturedCardsSection`, so I've added it to "Sections" board.

| Sections | Typography | Buttons | Components & Elements |
| --------------- | --------------- | --------------- | --------------- |
|  Screenshot 2024-09-27 at 18 54 25 |  Screenshot 2024-09-27 at 18 54 08 | Screenshot 2024-09-30 at 18 31 38 | Screenshot 2024-09-27 at 18 54 35 |

### CODE REVIEW

**Comment by tetianaman** [comment]
> What has happened here?

Do you mean Display section? I removed it. They all are headings. I don't see a sense to separate one.

**Comment by alisey** [comment]
> Do you mean Display section? I removed it. They all are headings. I don't see a sense to separate one.

What does "Sections" refer to?

**Comment by tetianaman** [comment]
> What does "Sections" refer to? 

These are sections. Somebody named class for them and I kept it. 
Screenshot 2024-09-26 at 18 49 10

> Why does it suddenly say "Core components"? Headings are not even components.
Right. Hm ... What to write there? To not duplicate ui kit title under the line

**Comment by tetianaman** [comment]
Maybe we can move `title` into the header:
UI Kit | Typography
---------------------
UI Kit | Sections
---------------------
etc

**Comment by tetianaman** [comment]
> Why did you use component names in PascalCase?
This is not component name. This is `Variant` name - how it will appear in Add Panel. It seemed to me ok to name in PascalCase as components. I see that we have separate names in Add Panel too: "Text Input" etc. 
I can rename.

**Comment by tetianaman** [comment]
> What does "Sections" refer to?

I got what you mean! It's a mistake, "Typography" word should be there.

File: _codux/ui-kit/typography/typography.board.tsx [hunk 1]
```diff
@@ -1,89 +1,105 @@
...
-                <span className={styles.uikit}>UI Kit</span>
-                <span className={styles.foundation}> | Foundation</span>
-                <hr className={styles.hrSolid} />
-                <h3 className={styles.sectionTitle}>Typography</h3>
-                <h4 className={styles.sectionHeader}>DISPLAY</h4>
+        <div className="uiKitContainer">
+            <div className="uiKitHeader">
+                <b>UI Kit</b> | Core Components
+            </div>
+            <h3 className="uiKitTitle">Sections</h3>
```
Comment (tetianaman) [inline _codux/ui-kit/typography/typography.board.tsx:11]: Typography

File: _codux/ui-kit/typography/typography.board.tsx [hunk 2]
```diff
@@ -1,89 +1,105 @@
...
-        <div className={styles.container}>
-            <div>
-                <span className={styles.uikit}>UI Kit</span>
-                <span className={styles.foundation}> | Foundation</span>
-                <hr className={styles.hrSolid} />
-                <h3 className={styles.sectionTitle}>Typography</h3>
-                <h4 className={styles.sectionHeader}>DISPLAY</h4>
+        <div className="uiKitContainer">
+            <div className="uiKitHeader">
+                <b>UI Kit</b> | Core Components
```
Comment (tetianaman) [inline _codux/ui-kit/typography/typography.board.tsx:9]: UI Kit | Foundation

**Comment by alisey** [comment]
There's still a lot of excessive / repeated stuff. It could be simplified further into something like:
```jsx
import Kit from '_codux/ui-kit-utils';

<Kit category="Foundation" title="Typography">
    <Kit.Section title="Headings">
        <Kit.Subsection title="Number Input">
            <NumberInput />
        </Kit.Subsection>
    </Kit.Section>
</Kit>
```

**Comment by tetianaman** [comment]
> And it would be automatically large because or <Kit responsive> wrapping it, which would set the base font size for all nested <Kit.*> elements to something like max(14px, 2vw). 

It's not quite clear for me if you suggested to pass some `responsive` prop or used it to indicate that `Kit` is responsive. I did without `responsive` prop. `Kit` font sizes and margins, paddings, some width are responsive. Works ok, however often good look depends on zoom value. 

> Without affecting font size of the content.

If content has no defined own font styles, the ui kit section font styles affects it.

**Comment by tetianaman** [comment]
Draft: one more self-review needed with a fresh mind

File: _codux/ui-kit/buttons/buttons.board.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import { ChevronRightIcon } from '../../../src/components/icons/chevron-right-icon';
+import { ArrowRightIcon } from '../../../src/components/icons/arrow-right-icon';
+import { Kit } from '../ui-kit-utils/kit';
+
+import styles from './buttons.board.module.scss';
+
+export default createBoard({
+    name: 'Buttons',
+    Board: () => (
+        <Kit category="Core Components" title="Buttons">
```
Comment (Letushev) [inline _codux/ui-kit/buttons/buttons.board.tsx:17]: It wraps everthing with the "Buttons" title, but there are not only buttons here.

File: _codux/ui-kit/buttons/buttons.board.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                        <LabelWithArrow>Shop Now</LabelWithArrow>
+                    </Variant>
+                    <Kit.Description>Label With Arrow</Kit.Description>
+                </Kit.Item>
+            </Kit.Section>
+
+            <Kit.Section title="Icons">
+                <div className={styles.itemsGroup}>
+                    <Kit.Item>
+                        <div className={styles.icon}>
```
Comment (Letushev) [inline _codux/ui-kit/buttons/buttons.board.tsx:48]: I guess we can put className on the <Kit.Item>?

File: _codux/ui-kit/components/components.board.tsx [hunk 5]
```diff
@@ -2,69 +2,96 @@ import { createBoard, Variant } from '@wixc3/react-board';
...
+                        <Variant name="Product Card">
+                            <ProductCard
+                                name="Bamboo Toothbrush"
+                                imageUrl="https://static.wixstatic.com/media/c837a6_18152edaef9940ca88f446ae94b48a47~mv2.jpg/v1/fill/w_824,h_1098,al_c,q_85,usm_0.66_1.00_0.01,enc_auto/c837a6_18152edaef9940ca88f446ae94b48a47~mv2.jpg"
+                                ribbon="NEW"
+                                price="$6"
+                                discountedPrice="$5.5"
+                            />
+                        </Variant>
+                        <Kit.Description>Product Card</Kit.Description>
```
Comment (Letushev) [inline _codux/ui-kit/components/components.board.tsx:71]: It looks as if "Product Card" is part of the card. Can we put the kit description on top and with a distinctive styling?

Screenshot 2024-09-30 at 14 28 04

IMO it's doesn't look good in sections too.

Screenshot 2024-09-30 at 14 30 42

File: _codux/ui-kit/ui-kit-utils/kit.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    className?: string;
+}
+
+export const Kit = ({ category, title, children, className }: KitProps) => {
+    return (
+        <div className={classNames(styles.container, className)}>
+            <div className={styles.header}>
+                <b>UI Kit</b> | {category}
+            </div>
+            <h3 className={styles.title}>{title}</h3>
```
Comment (Letushev) [inline _codux/ui-kit/ui-kit-utils/kit.tsx:18]: Itn't category already a title?

File: _codux/ui-kit/typography/typography.board.tsx [hunk 7]
```diff
@@ -1,89 +1,98 @@
...
+                        </div>
+                    </Variant>
+                    <Kit.Description>
+                        <b>--paragraph1:</b> Figtree (400) / 16px / 1.5
+                    </Kit.Description>
+                </Kit.Item>
+                <Kit.Item>
+                    <Variant name="Paragraph2">
+                        <div className="paragraph2">
+                            We ignite opportunity by setting the world in motion. 0123456789
```
Comment (Letushev) [inline _codux/ui-kit/typography/typography.board.tsx:73]: Can we remove this `0123456789`?

File: _codux/ui-kit/buttons/buttons.board.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+                    </Kit.Item>
+                </div>
+            </Kit.Section>
+        </Kit>
+    ),
+    environmentProps: {
+        windowWidth: 427,
+        windowHeight: 704,
+    },
+    isSnippet: true,
```
Comment (Letushev) [inline _codux/ui-kit/buttons/buttons.board.tsx:140]: I think it should'b be a snippet as well as others ui kit boards.

File: _codux/ui-kit/ui-kit-utils/kit.tsx (hunk 6 above)
Comment (tetianaman) [reply _codux/ui-kit/ui-kit-utils/kit.tsx:18]: Categories are 'Foundation' and 'Core components'. 
Titles are 'Typography', 'Buttons', 'Sections', 'Components & Elements'

File: _codux/ui-kit/typography/typography.board.tsx (hunk 7 above)
Comment (tetianaman) [reply _codux/ui-kit/typography/typography.board.tsx:73]: It's by design to demonstrate how text styling looks for numbers.

File: _codux/ui-kit/buttons/buttons.board.tsx (hunk 4 above)
Comment (tetianaman) [reply _codux/ui-kit/buttons/buttons.board.tsx:48]: It has `className`, but I need additional `div`. 
Item here is `flex` column, `div.icon` is the wrapper where I center an icon.

File: _codux/ui-kit/ui-kit-utils/kit.tsx (hunk 6 above)
Comment (Letushev) [reply _codux/ui-kit/ui-kit-utils/kit.tsx:18]: > Titles are 'Typography', 'Buttons', 'Sections', 'Components & Elements'

Aren't they titles of `<Kit.Section>`?

Comment (tetianaman) [reply _codux/ui-kit/ui-kit-utils/kit.tsx:18]: Nope. 
Sometimes they are the same however.

File: _codux/ui-kit/buttons/buttons.board.tsx (hunk 3 above)
Comment (Letushev) [reply _codux/ui-kit/buttons/buttons.board.tsx]: Buttons and Icons

File: _codux/ui-kit/components/components.board.tsx (hunk 5 above)
Comment (Dana-Levy) [reply _codux/ui-kit/components/components.board.tsx:71]: If there's only one component under the section, we can remove the description,
if there's more than one, we keep it.  

Note: This is a temporary solution - @RanBerkovich1  has a task that gives each variant a name automatically

File: _codux/ui-kit/buttons/buttons.board.tsx (hunk 3 above)
Comment (tetianaman) [reply _codux/ui-kit/buttons/buttons.board.tsx]: @Letushev Dana agreed.

**Review by Letushev — APPROVED** [review]

**Pull request merged**